<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵 | Lihuax's Blog</title><meta name="author" content="Lihuax"><meta name="copyright" content="Lihuax"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="❗ Topic时空组学数据基本处理流程与原理-从reads到表达矩阵 Stereo-seq技术基础原理文件 mask文件: 记录CID序列与空间位置的文件 image文件: 图像质控结果文件 Fastq文件: 转录粗测序数据文件  芯片 芯片大小: 1cm*1cm 纳米球尺寸: 直径220nm 纳米球间距: 500nm spot数量: $4\times10^8$个 barcode长度: 25bp,">
<meta property="og:type" content="article">
<meta property="og:title" content="时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵">
<meta property="og:url" content="https://zehua0417.github.io/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/1_%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E5%9F%BA%E6%9C%AC%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B%E4%B8%8E%E5%8E%9F%E7%90%86-%E4%BB%8Ereads%E5%88%B0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5/index.html">
<meta property="og:site_name" content="Lihuax&#39;s Blog">
<meta property="og:description" content="❗ Topic时空组学数据基本处理流程与原理-从reads到表达矩阵 Stereo-seq技术基础原理文件 mask文件: 记录CID序列与空间位置的文件 image文件: 图像质控结果文件 Fastq文件: 转录粗测序数据文件  芯片 芯片大小: 1cm*1cm 纳米球尺寸: 直径220nm 纳米球间距: 500nm spot数量: $4\times10^8$个 barcode长度: 25bp,">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://zehua0417.github.io/img/avatar.png">
<meta property="article:published_time" content="2024-07-23T10:02:26.063Z">
<meta property="article:modified_time" content="2024-07-23T10:02:26.063Z">
<meta property="article:author" content="Lihuax">
<meta property="article:tag" content="BGI">
<meta property="article:tag" content="bio">
<meta property="article:tag" content="bioinformatics">
<meta property="article:tag" content="single-cell">
<meta property="article:tag" content="Course_note">
<meta property="article:tag" content="stereo">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://zehua0417.github.io/img/avatar.png"><link rel="shortcut icon" href="/img/favicon.png"><link rel="canonical" href="https://zehua0417.github.io/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/1_%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E5%9F%BA%E6%9C%AC%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B%E4%B8%8E%E5%8E%9F%E7%90%86-%E4%BB%8Ereads%E5%88%B0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css" media="print" onload="this.media='all'"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.min.css" media="print" onload="this.media='all'"><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  noticeOutdate: undefined,
  highlight: {"plugin":"highlighjs","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false},
  copy: {
    success: 'Copy successfully',
    error: 'Copy error',
    noSupport: 'The browser does not support'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: 'Just',
    min: 'minutes ago',
    hour: 'hours ago',
    day: 'days ago',
    month: 'months ago'
  },
  copyright: undefined,
  lightbox: 'fancybox',
  Snackbar: undefined,
  source: {
    justifiedGallery: {
      js: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.js',
      css: 'https://cdn.jsdelivr.net/npm/flickr-justified-gallery/dist/fjGallery.min.css'
    }
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  postUpdate: '2024-07-23 10:02:26'
}</script><noscript><style type="text/css">
  #nav {
    opacity: 1
  }
  .justified-gallery img {
    opacity: 1
  }

  #recent-posts time,
  #post-meta time {
    display: inline !important
  }
</style></noscript><script>(win=>{
    win.saveToLocal = {
      set: function setWithExpiry(key, value, ttl) {
        if (ttl === 0) return
        const now = new Date()
        const expiryDay = ttl * 86400000
        const item = {
          value: value,
          expiry: now.getTime() + expiryDay,
        }
        localStorage.setItem(key, JSON.stringify(item))
      },

      get: function getWithExpiry(key) {
        const itemStr = localStorage.getItem(key)

        if (!itemStr) {
          return undefined
        }
        const item = JSON.parse(itemStr)
        const now = new Date()

        if (now.getTime() > item.expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return item.value
      }
    }
  
    win.getScript = url => new Promise((resolve, reject) => {
      const script = document.createElement('script')
      script.src = url
      script.async = true
      script.onerror = reject
      script.onload = script.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        script.onload = script.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(script)
    })
  
    win.getCSS = (url,id = false) => new Promise((resolve, reject) => {
      const link = document.createElement('link')
      link.rel = 'stylesheet'
      link.href = url
      if (id) link.id = id
      link.onerror = reject
      link.onload = link.onreadystatechange = function() {
        const loadState = this.readyState
        if (loadState && loadState !== 'loaded' && loadState !== 'complete') return
        link.onload = link.onreadystatechange = null
        resolve()
      }
      document.head.appendChild(link)
    })
  
      win.activateDarkMode = function () {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      win.activateLightMode = function () {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }
      const t = saveToLocal.get('theme')
    
          if (t === 'dark') activateDarkMode()
          else if (t === 'light') activateLightMode()
        
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        if (asideStatus === 'hide') {
          document.documentElement.classList.add('hide-aside')
        } else {
          document.documentElement.classList.remove('hide-aside')
        }
      }
    
    const detectApple = () => {
      if(/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)){
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
    })(window)</script><meta name="generator" content="Hexo 6.3.0"></head><body><div id="sidebar"><div id="menu-mask"></div><div id="sidebar-menus"><div class="avatar-img is-center"><img src="/img/avatar.png" onerror="onerror=null;src='/img/friend_404.gif'" alt="avatar"/></div><div class="sidebar-site-data site-data is-center"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">11</div></a><a href="/tags/"><div class="headline">Tags</div><div class="length-num">16</div></a><a href="/categories/"><div class="headline">Categories</div><div class="length-num">2</div></a></div><hr class="custom-hr"/><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div></div></div></div><div class="post" id="body-wrap"><header class="post-bg" id="page-header" style="background-image: url('/img/top_img.png')"><nav id="nav"><span id="blog-info"><a href="/" title="Lihuax's Blog"><span class="site-name">Lihuax's Blog</span></a></span><div id="menus"><div class="menus_items"><div class="menus_item"><a class="site-page" href="/"><i class="fa-fw fas fa-home"></i><span> Home</span></a></div><div class="menus_item"><a class="site-page" href="/archives/"><i class="fa-fw fas fa-archive"></i><span> Archives</span></a></div><div class="menus_item"><a class="site-page" href="/tags/"><i class="fa-fw fas fa-tags"></i><span> Tags</span></a></div><div class="menus_item"><a class="site-page" href="/categories/"><i class="fa-fw fas fa-folder-open"></i><span> Categories</span></a></div></div><div id="toggle-menu"><a class="site-page" href="javascript:void(0);"><i class="fas fa-bars fa-fw"></i></a></div></div></nav><div id="post-info"><h1 class="post-title">时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2024-07-23T10:02:26.063Z" title="Created 2024-07-23 10:02:26">2024-07-23</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2024-07-23T10:02:26.063Z" title="Updated 2024-07-23 10:02:26">2024-07-23</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title="时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵"><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post View:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="post-content" id="article-container"><h1 id="❗-Topic"><a href="#❗-Topic" class="headerlink" title="❗ Topic"></a>❗ Topic</h1><p><strong>时空组学数据基本处理流程与原理-从reads到表达矩阵</strong></p>
<h2 id="Stereo-seq技术基础原理"><a href="#Stereo-seq技术基础原理" class="headerlink" title="Stereo-seq技术基础原理"></a>Stereo-seq技术基础原理</h2><h3 id="文件"><a href="#文件" class="headerlink" title="文件"></a>文件</h3><ul>
<li><strong>mask文件</strong>: 记录CID序列与空间位置的文件</li>
<li><strong>image文件</strong>: 图像质控结果文件</li>
<li><strong>Fastq文件</strong>: 转录粗测序数据文件</li>
</ul>
<h3 id="芯片"><a href="#芯片" class="headerlink" title="芯片"></a>芯片</h3><ul>
<li>芯片大小: 1cm*1cm</li>
<li>纳米球尺寸: 直径220nm</li>
<li>纳米球间距: 500nm</li>
<li>spot数量: $4\times10^8$个</li>
<li>barcode长度: 25bp, $4^{25}种$</li>
<li>UMI长度: 10bp</li>
<li>CID: 25bp</li>
</ul>
<h2 id="数据分析SAW-STOmics-analysis-workflow-整体流程介绍"><a href="#数据分析SAW-STOmics-analysis-workflow-整体流程介绍" class="headerlink" title="数据分析SAW(STOmics analysis workflow)整体流程介绍"></a>数据分析SAW(STOmics analysis workflow)整体流程介绍</h2><p><a target="_blank" rel="noopener" href="https://github.com/BGIResearch/SAW">SAW github link</a></p>
<h3 id="Stereo-seq"><a href="#Stereo-seq" class="headerlink" title="Stereo-seq"></a>Stereo-seq</h3><p>SAW 处理 Stereo-seq 的测序数据以生成空间基因表达矩阵，用户可以将这些文件作为起点进行下游分析。SAW 包含十三个基本和建议的流程和辅助工具，以支持其他便捷功能。</p>
<h4 id="输入文件"><a href="#输入文件" class="headerlink" title="输入文件"></a>输入文件</h4><h5 id="reference"><a href="#reference" class="headerlink" title="reference"></a>reference</h5><ul>
<li><strong>输入文件</strong><ul>
<li>基因组序列文件: genome.fa</li>
<li>基因注释文件: genes.gtf$&#x2F;$gff<figure class="highlight txt"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">.</span><br><span class="line">└── specieName</span><br><span class="line">    ├── STAR_SJ100</span><br><span class="line">    │   ├── Genome</span><br><span class="line">    │   ├── SA</span><br><span class="line">    │   ├── SAindex</span><br><span class="line">    │   ├── chrLength.txt</span><br><span class="line">    │   ├── chrName.txt</span><br><span class="line">    │   ├── chrNameLength.txt</span><br><span class="line">    │   ├── chrStart.txt</span><br><span class="line">    │   ├── exonGeTrInfo.tab</span><br><span class="line">    │   ├── exonInfo.tab</span><br><span class="line">    │   ├── geneInfo.tab</span><br><span class="line">    │   ├── genomeParameters.txt</span><br><span class="line">    │   ├── sjdbInfo.txt</span><br><span class="line">    │   ├── sjdbList.fromGTF.out.tab</span><br><span class="line">    │   ├── sjdbList.out.tab</span><br><span class="line">    │   └── transcriptInfo.tab</span><br><span class="line">    ├── genes</span><br><span class="line">    │   └── genes.gtf</span><br><span class="line">    └── genome</span><br><span class="line">        └── genome.fa</span><br><span class="line"></span><br><span class="line">4 directories, 17 files</span><br></pre></td></tr></table></figure></li>
</ul>
</li>
<li><strong>输出文件</strong><ul>
<li>参考基因组索引目录: &#x2F;path&#x2F;to&#x2F;genomeDir</li>
</ul>
</li>
<li><strong>检查索引文件的小工具</strong><ul>
<li>SAW: checkGTF</li>
</ul>
</li>
<li>参考脚本: <a target="_blank" rel="noopener" href="https://github.com/STOmics/SAW/tree/c6a058239d944a427278ee262008d1828a96b13f/Scripts/pre_buildIndexedRef">example script</a></li>
</ul>
<h5 id="mask"><a href="#mask" class="headerlink" title="mask"></a>mask</h5><ul>
<li><strong>格式</strong>: $*.h5$</li>
</ul>
<h5 id="image"><a href="#image" class="headerlink" title="image"></a>image</h5><ul>
<li><strong>格式</strong>: $SN_^*.ipr$&amp;$SN_^*.tar.gz$</li>
</ul>
<h5 id="fastaq"><a href="#fastaq" class="headerlink" title="fastaq"></a>fastaq</h5><p><strong>PE format</strong></p>
<ul>
<li>read1 &#x3D; CID + MID</li>
<li>read2 &#x3D; mRNA</li>
<li>不需要splitMask<br><strong>SE format</strong></li>
<li>read_name &#x3D; CID + MID</li>
<li>read &#x3D; mRNA</li>
<li>需要splitMask</li>
</ul>
<h4 id="流程"><a href="#流程" class="headerlink" title="流程"></a>流程</h4><h5 id="mapping-比对"><a href="#mapping-比对" class="headerlink" title="mapping 比对"></a>mapping 比对</h5><ul>
<li><strong>Step1 BarcodeMapping</strong>: 原始测序读取（存储在FASTQ文件中）的CID与Stereo-seq Chip T MASK文件中保存的CID-坐标键值对记录进行匹配（允许1个错配）。根据MASK文件中的记录，为可以配对的读取添加坐标信息。获得坐标注释的读取被称为有效CID mRNA读取（有效CID读取）。</li>
<li><strong>Step2 Filtering</strong>: <ul>
<li>丢弃不合格的MID读取（不满足进一步分析要求）</li>
<li>过滤含有接头adapter的读取</li>
<li>过滤长度(去除poly-A后)小于30的短读取</li>
</ul>
</li>
<li><strong>Step3 GenomeMapping</strong>: mRNA比对到参考物种基因组<br><strong>输出文件</strong>:</li>
<li><strong>BAM格式比对结果文件</strong>: <ul>
<li>$*.Aligned.sortedByCoord.out.bam$</li>
<li>tag列包含每条reads的空间位置信息</li>
</ul>
</li>
<li>比对结果数据统计文件<ul>
<li>$^*_barcodeMap.stat$: BarcodeMapping信息</li>
<li>$^*.Log.final.out$: GenomeMapping信息</li>
<li>$^*.barcodeReadsCount.txt$: 每个barcode的reads数</li>
</ul>
</li>
</ul>
<h5 id="merge-合并"><a href="#merge-合并" class="headerlink" title="merge 合并"></a>merge 合并</h5><h5 id="count-注释"><a href="#count-注释" class="headerlink" title="count 注释"></a>count 注释</h5><table>
<thead>
<tr>
<th>注释类型</th>
<th>注释逻辑</th>
</tr>
</thead>
<tbody><tr>
<td>Intron类型</td>
<td>reads与intron有超过50%的overlap</td>
</tr>
<tr>
<td>antisense统计</td>
<td>比对上任意反链的gene</td>
</tr>
<tr>
<td>transcriptome统计</td>
<td>注释上exon或intron且同链</td>
</tr>
<tr>
<td><strong>输出文件</strong></td>
<td></td>
</tr>
</tbody></table>
<ul>
<li><strong>BAM格式比对结果文件与GEF格式表达矩阵</strong><ul>
<li>$^*.Aligned.sortedByCoord.out.merge.q10.deduo.target.bam$, Tag列新增了注释成功的基因信息</li>
<li>$^*.raw.gef$, 基因表达矩阵</li>
</ul>
</li>
<li><strong>日志文件</strong><ul>
<li>$logs&#x2F;Bam2Gem_data_time.log$, 记录了被跳过的基因</li>
</ul>
</li>
</ul>
<h5 id="register-rapidRegister-图像-配准"><a href="#register-rapidRegister-图像-配准" class="headerlink" title="register$&#x2F;$rapidRegister (图像)配准"></a>register$&#x2F;$rapidRegister (图像)配准</h5><p>基于空间位置信息, 将影像图最优映射到基因表达矩阵上<br>配准逻辑:</p>
<ul>
<li>基于TrackLine配准<ul>
<li>选模板(scale &amp; rotation)</li>
<li>根据模板推全局</li>
</ul>
</li>
<li>算法原理<ul>
<li>重心法</li>
</ul>
</li>
<li>修正&#x2F;微调<ul>
<li>根据TrackLine进行微调</li>
</ul>
</li>
</ul>
<h5 id="tissueCut-组织分隔"><a href="#tissueCut-组织分隔" class="headerlink" title="tissueCut 组织分隔"></a>tissueCut 组织分隔</h5><p>在影像图上获取组织边界</p>
<ul>
<li>有ssDNA图像: 基于影像图分隔</li>
<li>无ssDNA图像: 基于表达矩阵分隔<br>基于组织分隔掩模图, 结合矩阵进行组织区域表达矩阵提取, 输出SN.tissue.gef</li>
</ul>
<h5 id="spatialCluster-空间聚类"><a href="#spatialCluster-空间聚类" class="headerlink" title="spatialCluster 空间聚类"></a>spatialCluster 空间聚类</h5><p>筛选高变菊英后进行PCA和UMAP降维, 再使用Leiden等算法进行聚类分析</p>
<h5 id="saturation-饱和度计算"><a href="#saturation-饱和度计算" class="headerlink" title="saturation 饱和度计算"></a>saturation 饱和度计算</h5><p>根据饱和度曲线判断数据是否需要加测<br>$$sequence<del>saturation&#x3D;1-\frac{Unique</del>Reads}{Total Reads}$$</p>
<h5 id="report-报告输出"><a href="#report-报告输出" class="headerlink" title="report 报告输出"></a>report 报告输出</h5><div class="gallery">
    <div class="fj-gallery  data" data-rowHeight="220" data-limit="10">
    <span class="gallery-data">[{"url":"/img/obsimg/SAW_v7.1_workflow.png","alt":"img"}]</span>
    </div>
    <button class="gallery-load-more"><span>Load More</span><i class="fa-solid fa-arrow-down"></i></button>
    </div>
<h3 id="蛋白质组学和转录组学"><a href="#蛋白质组学和转录组学" class="headerlink" title="蛋白质组学和转录组学"></a>蛋白质组学和转录组学</h3><p>SAW 蛋白质组学和转录组学工作流程处理来自单芯片的测序数据以生成空间基因和蛋白质表达矩阵。您可以使用这两个组学信息开始更高维度的研究。此工作流程包括 23 个基本和建议流程。</p>
<div class="gallery">
    <div class="fj-gallery  data" data-rowHeight="220" data-limit="10">
    <span class="gallery-data">[{"url":"/img/obsimg/SAW_pt_v7.1_workflow.png|400","alt":"img"}]</span>
    </div>
    <button class="gallery-load-more"><span>Load More</span><i class="fa-solid fa-arrow-down"></i></button>
    </div>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta">Author: </span><span class="post-copyright-info"><a href="https://zehua0417.github.io">Lihuax</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta">Link: </span><span class="post-copyright-info"><a href="https://zehua0417.github.io/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/1_%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E5%9F%BA%E6%9C%AC%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B%E4%B8%8E%E5%8E%9F%E7%90%86-%E4%BB%8Ereads%E5%88%B0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5/">https://zehua0417.github.io/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/1_%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E5%9F%BA%E6%9C%AC%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B%E4%B8%8E%E5%8E%9F%E7%90%86-%E4%BB%8Ereads%E5%88%B0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta">Copyright Notice: </span><span class="post-copyright-info">All articles in this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless stating additionally.</span></div></div><div class="tag_share"><div class="post-meta__tag-list"><a class="post-meta__tags" href="/tags/BGI/">BGI</a><a class="post-meta__tags" href="/tags/bio/">bio</a><a class="post-meta__tags" href="/tags/bioinformatics/">bioinformatics</a><a class="post-meta__tags" href="/tags/single-cell/">single-cell</a><a class="post-meta__tags" href="/tags/Course-note/">Course_note</a><a class="post-meta__tags" href="/tags/stereo/">stereo</a></div><div class="post_share"><div class="social-share" data-image="/img/avatar.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><div class="prev-post pull-left"><a href="/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E5%8D%95%E7%BB%86%E8%83%9E%E5%88%86%E6%9E%90%E6%8F%AD%E7%A4%BA%E7%BB%93%E7%9B%B4%E8%82%A0%E7%99%8C%E5%85%8D%E7%96%AB%E6%B2%BB%E7%96%97%E4%B8%8D%E5%90%8C%E5%8F%8D%E5%BA%94%E7%9A%84%E7%BB%86%E8%83%9E%E5%8A%A8%E6%80%81/" title="时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">Previous Post</div><div class="prev_info">时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer</div></div></a></div><div class="next-post pull-right"><a href="/2023/11/07/%E5%AD%A6%E4%B9%A0/%E4%B8%AA%E4%BA%BAbashscript%E8%84%9A%E6%9C%AC%E5%88%86%E4%BA%AB/" title="个人bashscript脚本分享"><div class="cover" style="background: var(--default-bg-color)"></div><div class="pagination-info"><div class="label">Next Post</div><div class="next_info">个人bashscript脚本分享</div></div></a></div></nav><div class="relatedPosts"><div class="headline"><i class="fas fa-thumbs-up fa-fw"></i><span>Related Articles</span></div><div class="relatedPosts-list"><div><a href="/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E5%8D%95%E7%BB%86%E8%83%9E%E5%88%86%E6%9E%90%E6%8F%AD%E7%A4%BA%E7%BB%93%E7%9B%B4%E8%82%A0%E7%99%8C%E5%85%8D%E7%96%AB%E6%B2%BB%E7%96%97%E4%B8%8D%E5%90%8C%E5%8F%8D%E5%BA%94%E7%9A%84%E7%BB%86%E8%83%9E%E5%8A%A8%E6%80%81/" title="时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer"><div class="cover" style="background: var(--default-bg-color)"></div><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2024-07-23</div><div class="title">时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer</div></div></a></div><div><a href="/2023/08/06/%E5%AD%A6%E4%B9%A0/%E7%90%86%E8%A7%A3%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E9%98%85%E8%AF%BB%E7%AC%94%E8%AE%B0/" title="&lt;&lt;理解生物信息学&gt;&gt; 阅读笔记 第二部分 序列联配"><div class="cover" style="background: var(--default-bg-color)"></div><div class="content is-center"><div class="date"><i class="far fa-calendar-alt fa-fw"></i> 2023-08-06</div><div class="title">&lt;&lt;理解生物信息学&gt;&gt; 阅读笔记 第二部分 序列联配</div></div></a></div></div></div><hr class="custom-hr"/><div id="post-comment"><div class="comment-head"><div class="comment-headline"><i class="fas fa-comments fa-fw"></i><span> Comment</span></div></div><div class="comment-wrap"><div><div class="vcomment" id="vcomment"></div></div></div></div></div><div class="aside-content" id="aside-content"><div class="card-widget card-info"><div class="is-center"><div class="avatar-img"><img src="/img/avatar.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info__name">Lihuax</div><div class="author-info__description">lov(life oriented living)</div></div><div class="card-info-data site-data is-center"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">11</div></a><a href="/tags/"><div class="headline">Tags</div><div class="length-num">16</div></a><a href="/categories/"><div class="headline">Categories</div><div class="length-num">2</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/zehua0417/zehua0417.github.io.git"><i class="fab fa-github"></i><span>Follow Me</span></a><div class="card-info-social-icons is-center"><a class="social-icon" href="https://github.com/zehua0417" target="_blank" title="Github"><i class="fab fa-github" style="color: #24292e;"></i></a><a class="social-icon" href="mailto:zehuali0417@gmail.com" target="_blank" title="Email"><i class="fas fa-envelope" style="color: #4a7dbe;"></i></a></div></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>Announcement</span></div><div class="announcement_content">最新加入评论功能(基于valine),更新follow me 为仓库地址, 修复了一些bug, 优化了一些细节,由于不支持mardown部分语法, 部分网页图片出错正在修复,...</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>Catalog</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E2%9D%97-Topic"><span class="toc-number">1.</span> <span class="toc-text">❗ Topic</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Stereo-seq%E6%8A%80%E6%9C%AF%E5%9F%BA%E7%A1%80%E5%8E%9F%E7%90%86"><span class="toc-number">1.1.</span> <span class="toc-text">Stereo-seq技术基础原理</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%96%87%E4%BB%B6"><span class="toc-number">1.1.1.</span> <span class="toc-text">文件</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E8%8A%AF%E7%89%87"><span class="toc-number">1.1.2.</span> <span class="toc-text">芯片</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90SAW-STOmics-analysis-workflow-%E6%95%B4%E4%BD%93%E6%B5%81%E7%A8%8B%E4%BB%8B%E7%BB%8D"><span class="toc-number">1.2.</span> <span class="toc-text">数据分析SAW(STOmics analysis workflow)整体流程介绍</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#Stereo-seq"><span class="toc-number">1.2.1.</span> <span class="toc-text">Stereo-seq</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#%E8%BE%93%E5%85%A5%E6%96%87%E4%BB%B6"><span class="toc-number">1.2.1.1.</span> <span class="toc-text">输入文件</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#reference"><span class="toc-number">1.2.1.1.1.</span> <span class="toc-text">reference</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#mask"><span class="toc-number">1.2.1.1.2.</span> <span class="toc-text">mask</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#image"><span class="toc-number">1.2.1.1.3.</span> <span class="toc-text">image</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#fastaq"><span class="toc-number">1.2.1.1.4.</span> <span class="toc-text">fastaq</span></a></li></ol></li><li class="toc-item toc-level-4"><a class="toc-link" href="#%E6%B5%81%E7%A8%8B"><span class="toc-number">1.2.1.2.</span> <span class="toc-text">流程</span></a><ol class="toc-child"><li class="toc-item toc-level-5"><a class="toc-link" href="#mapping-%E6%AF%94%E5%AF%B9"><span class="toc-number">1.2.1.2.1.</span> <span class="toc-text">mapping 比对</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#merge-%E5%90%88%E5%B9%B6"><span class="toc-number">1.2.1.2.2.</span> <span class="toc-text">merge 合并</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#count-%E6%B3%A8%E9%87%8A"><span class="toc-number">1.2.1.2.3.</span> <span class="toc-text">count 注释</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#register-rapidRegister-%E5%9B%BE%E5%83%8F-%E9%85%8D%E5%87%86"><span class="toc-number">1.2.1.2.4.</span> <span class="toc-text">register$&#x2F;$rapidRegister (图像)配准</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#tissueCut-%E7%BB%84%E7%BB%87%E5%88%86%E9%9A%94"><span class="toc-number">1.2.1.2.5.</span> <span class="toc-text">tissueCut 组织分隔</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#spatialCluster-%E7%A9%BA%E9%97%B4%E8%81%9A%E7%B1%BB"><span class="toc-number">1.2.1.2.6.</span> <span class="toc-text">spatialCluster 空间聚类</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#saturation-%E9%A5%B1%E5%92%8C%E5%BA%A6%E8%AE%A1%E7%AE%97"><span class="toc-number">1.2.1.2.7.</span> <span class="toc-text">saturation 饱和度计算</span></a></li><li class="toc-item toc-level-5"><a class="toc-link" href="#report-%E6%8A%A5%E5%91%8A%E8%BE%93%E5%87%BA"><span class="toc-number">1.2.1.2.8.</span> <span class="toc-text">report 报告输出</span></a></li></ol></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E8%9B%8B%E7%99%BD%E8%B4%A8%E7%BB%84%E5%AD%A6%E5%92%8C%E8%BD%AC%E5%BD%95%E7%BB%84%E5%AD%A6"><span class="toc-number">1.2.2.</span> <span class="toc-text">蛋白质组学和转录组学</span></a></li></ol></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>Recent Post</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E5%8D%95%E7%BB%86%E8%83%9E%E5%88%86%E6%9E%90%E6%8F%AD%E7%A4%BA%E7%BB%93%E7%9B%B4%E8%82%A0%E7%99%8C%E5%85%8D%E7%96%AB%E6%B2%BB%E7%96%97%E4%B8%8D%E5%90%8C%E5%8F%8D%E5%BA%94%E7%9A%84%E7%BB%86%E8%83%9E%E5%8A%A8%E6%80%81/" title="时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer">时空组学经典论文阅读 Spatiotemporal single-cell analysis decodes cellular dynamics underlying different responses to immunotherapy in colorectal cancer</a><time datetime="2024-07-23T10:02:26.063Z" title="Created 2024-07-23 10:02:26">2024-07-23</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/07/23/%E5%AD%A6%E4%B9%A0/%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/1_%E6%97%B6%E7%A9%BA%E7%BB%84%E5%AD%A6%E6%95%B0%E6%8D%AE%E5%9F%BA%E6%9C%AC%E5%A4%84%E7%90%86%E6%B5%81%E7%A8%8B%E4%B8%8E%E5%8E%9F%E7%90%86-%E4%BB%8Ereads%E5%88%B0%E8%A1%A8%E8%BE%BE%E7%9F%A9%E9%98%B5/" title="时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵">时空专题1_时空组学数据基本处理流程与原理_从reads到表达矩阵</a><time datetime="2024-07-23T10:02:26.063Z" title="Created 2024-07-23 10:02:26">2024-07-23</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2023/11/07/%E5%AD%A6%E4%B9%A0/%E4%B8%AA%E4%BA%BAbashscript%E8%84%9A%E6%9C%AC%E5%88%86%E4%BA%AB/" title="个人bashscript脚本分享">个人bashscript脚本分享</a><time datetime="2023-11-07T00:00:00.000Z" title="Created 2023-11-07 00:00:00">2023-11-07</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2023/08/25/%E7%94%9F%E6%B4%BB/lihuax%E7%9A%84%E7%8E%8B%E5%B0%8F%E6%B3%A2%E6%9D%82%E6%96%87%E9%80%89/" title="lihuax的王小波杂文选">lihuax的王小波杂文选</a><time datetime="2023-08-25T17:30:00.000Z" title="Created 2023-08-25 17:30:00">2023-08-25</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2023/08/21/%E5%AD%A6%E4%B9%A0/%E5%9C%A8ubuntu%E9%BB%91%E7%AA%97%E5%8F%A3%E4%B8%AD%E9%85%8D%E7%BD%AEshadowsocks_%E9%80%82%E7%94%A8%E4%BA%8Ewsl/" title="在ubuntu黑窗口终端中配置shadowsocks(适用于wsl)">在ubuntu黑窗口终端中配置shadowsocks(适用于wsl)</a><time datetime="2023-08-21T00:00:00.000Z" title="Created 2023-08-21 00:00:00">2023-08-21</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2020 - 2024 By Lihuax</div><div class="framework-info"><span>Framework </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div><script type="text/javascript" id="maid-script" src="https://unpkg.com/mermaid@undefined/dist/mermaid.min.js?v=undefined"></script><script>if (window.mermaid) {
  var options = JSON.parse(document.getElementById('maid-script').getAttribute('mermaidoptioins'));
  mermaid.initialize(options);
}</script></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Read Mode"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="Switch Between Light And Dark Mode"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="Toggle between single-column and double-column"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside_config" type="button" title="Setting"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table Of Contents"><i class="fas fa-list-ul"></i></button><a id="to_comment" href="#post-comment" title="Scroll To Comments"><i class="fas fa-comments"></i></a><button id="go-up" type="button" title="Back To Top"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui/dist/fancybox/fancybox.umd.min.js"></script><div class="js-pjax"><script>if (!window.MathJax) {
  window.MathJax = {
    tex: {
      inlineMath: [['$', '$'], ['\\(', '\\)']],
      tags: 'ams'
    },
    chtml: {
      scale: 1.1
    },
    options: {
      renderActions: {
        findScript: [10, doc => {
          for (const node of document.querySelectorAll('script[type^="math/tex"]')) {
            const display = !!node.type.match(/; *mode=display/)
            const math = new doc.options.MathItem(node.textContent, doc.inputJax[0], display)
            const text = document.createTextNode('')
            node.parentNode.replaceChild(text, node)
            math.start = {node: text, delim: '', n: 0}
            math.end = {node: text, delim: '', n: 0}
            doc.math.push(math)
          }
        }, '']
      }
    }
  }
  
  const script = document.createElement('script')
  script.src = 'https://cdn.jsdelivr.net/npm/mathjax/es5/tex-mml-chtml.min.js'
  script.id = 'MathJax-script'
  script.async = true
  document.head.appendChild(script)
} else {
  MathJax.startup.document.state(0)
  MathJax.texReset()
  MathJax.typesetPromise()
}</script><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/katex/dist/katex.min.css"><script src="https://cdn.jsdelivr.net/npm/katex/dist/contrib/copy-tex.min.js"></script><script>(() => {
  document.querySelectorAll('#article-container span.katex-display').forEach(item => {
    btf.wrap(item, 'div', { class: 'katex-wrap'})
  })
})()</script><script>(() => {
  const $mermaid = document.querySelectorAll('#article-container .mermaid-wrap')
  if ($mermaid.length === 0) return
  const runMermaid = () => {
    window.loadMermaid = true
    const theme = document.documentElement.getAttribute('data-theme') === 'dark' ? 'dark' : 'default'

    Array.from($mermaid).forEach((item, index) => {
      const mermaidSrc = item.firstElementChild
      const mermaidThemeConfig = '%%{init:{ \'theme\':\'' + theme + '\'}}%%\n'
      const mermaidID = 'mermaid-' + index
      const mermaidDefinition = mermaidThemeConfig + mermaidSrc.textContent

      const renderFn = mermaid.render(mermaidID, mermaidDefinition)

      const renderV10 = () => {
        renderFn.then(({svg}) => {
          mermaidSrc.insertAdjacentHTML('afterend', svg)
        })
      }

      const renderV9 = svg => {
        mermaidSrc.insertAdjacentHTML('afterend', svg)
      }

      typeof renderFn === 'string' ? renderV9(renderFn) : renderV10()
    })
  }

  const loadMermaid = () => {
    window.loadMermaid ? runMermaid() : getScript('https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js').then(runMermaid)
  }

  btf.addModeChange('mermaid', runMermaid)

  window.pjax ? loadMermaid() : document.addEventListener('DOMContentLoaded', loadMermaid)
})()</script><script>function loadValine () {
  function initValine () {
    const valine = new Valine(Object.assign({
      el: '#vcomment',
      appId: 'g7ugLcLRZQImFitw8qLfg663-gzGzoHsz',
      appKey: 'KLsDFN1Oblr0vPS7zl3cg5wG',
      avatar: 'monsterid',
      serverURLs: 'https://g7uglclr.lc-cn-n1-shared.com',
      emojiMaps: "",
      path: window.location.pathname,
      visitor: false
    }, null))
  }

  if (typeof Valine === 'function') initValine() 
  else getScript('https://cdn.jsdelivr.net/npm/valine/dist/Valine.min.js').then(initValine)
}

if ('Valine' === 'Valine' || !true) {
  if (true) btf.loadComment(document.getElementById('vcomment'),loadValine)
  else setTimeout(loadValine, 0)
} else {
  function loadOtherComment () {
    loadValine()
  }
}</script></div><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>